src/js/components/PodInstancesTable.js

```diff
@@ -61,18 +61,46 @@ class PodInstancesTable extends React.Component {
     this.forceUpdate();
   }
 
+  componentWillReceiveProps(nextProps) {
+    let {checkedItems} = this.state;
+    let prevInstances = this.props.instances.getItems();
+    let nextInstances = nextProps.instances.getItems();
+
+    // When the `instances` property is changed and we have selected
+    // items, re-trigger selection change in order to remove checked
+    // entries that are no longer preset.
+
+    if (!deepEqual(prevInstances, nextInstances)) {
+      if (Object.keys(checkedItems).length) {
```

Comment:
We could swap those conditionals. I think checking `Object.keys(checkedItems).length` is faster than `deepEqual`